### PR TITLE
Magento channel buttons rearrangements #8022

### PR DESCRIPTION
**Pull request by vishu1993**

### CODE REVIEW

File: view/sale_channel_form.xml [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
     <xpath expr="/form/notebook/page[@id='configuration']/notebook/page[@id='connection']" position="inside">
-        <group id="magento_settings" states="{'invisible': Not(Eval('source') == 'magento')}">    
+        <group id="magento_settings" states="{'invisible': Not(Eval('source') == 'magento')}" colspan="8">    
             <label name="magento_url"/>
             <field name="magento_url" colspan="3"/>
             <label name="magento_api_user"/>
             <field name="magento_api_user"/>
             <label name="magento_api_key"/>
-            <field name="magento_api_key" widget="password"/>             
+            <field name="magento_api_key" widget="password"/>
```
Comment (tarunbhardwaj) [inline view/sale_channel_form.xml]: remove trailing space

File: view/sale_channel_form.xml [hunk 2]
```diff
@@ -36,6 +39,9 @@
     <xpath expr="/form/notebook/page[@id='configuration']/notebook/page[@id='taxes']" position="after">
         <page string="Carrier / Shipping Methods" id="carriers" states="{'invisible': Not(Eval('source') == 'magento')}">
             <field name="magento_carriers" colspan="4"/>
```
Comment (tarunbhardwaj) [inline view/sale_channel_form.xml:39]: states should be on this field also not on page

File: view/sale_channel_form.xml [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
             <label name="magento_url"/>
             <field name="magento_url" colspan="3"/>
             <label name="magento_api_user"/>
             <field name="magento_api_user"/>
             <label name="magento_api_key"/>
-            <field name="magento_api_key" widget="password"/>             
+            <field name="magento_api_key" widget="password"/>
+        </group>
+        <group id="configure_button" states="{'invisible': Not(Eval('source') == 'magento')}">
+            <button string="Configure Magento Connection" name="configure_magento_connection"/>
```
Comment (tarunbhardwaj) [inline view/sale_channel_form.xml]: Add states to button from python code.

File: view/sale_channel_form.xml [hunk 4]
```diff
@@ -34,8 +37,13 @@
         </group>
     </xpath>
     <xpath expr="/form/notebook/page[@id='configuration']/notebook/page[@id='taxes']" position="after">
-        <page string="Carrier / Shipping Methods" id="carriers" states="{'invisible': Not(Eval('source') == 'magento')}">
-            <field name="magento_carriers" colspan="4"/>
+        <page string="Carrier / Shipping Methods" id="carriers">
+            <group id="magento_carriers" states="{'invisible': Not(Eval('source') == 'magento')}">
+                <field name="magento_carriers" colspan="4"/>
```
Comment (tarunbhardwaj) [inline view/sale_channel_form.xml]: States should be in python code.

File: view/sale_channel_form.xml [hunk 5]
```diff
@@ -34,8 +37,13 @@
...
+            </group>
+            <group id="magento_buttons" col="3" colspan="6" states="{'invisible': Not(Eval('source') == 'magento')}">
+                <button string="Import Magento Carriers" name="import_magento_carriers"/>
```
Comment (tarunbhardwaj) [inline view/sale_channel_form.xml:41]: States should be in python code if its not already there.

File: channel.py [hunk 6]
```diff
@@ -123,8 +123,12 @@ def __setup__(cls):
                 'Current channel does not belongs to Magento !'
         })
         cls._buttons.update({
-            'import_magento_carriers': {},
-            'configure_magento_connection': {}
+            'import_magento_carriers': {
+                'invisible': ~(Eval('source') == 'magento')
```
Comment (prakashpp) [inline channel.py]: `Eval('source') != 'magento'` is more readable

File: channel.py [hunk 7]
```diff
@@ -123,8 +123,12 @@ def __setup__(cls):
...
+            },
+            'configure_magento_connection': {
+                'invisible': ~(Eval('source') == 'magento')
```
Comment (prakashpp) [inline channel.py]: `Eval('source') != 'magento'` is more readable

**Comment by vishu1993** [comment]
magento1

**Comment by vishu1993** [comment]
magento2

**Comment by vishu1993** [comment]
magento3

**Pull request merged**